cli/cmd/sources.go

```diff
@@ -330,6 +374,128 @@ var sourceStatusCmd = &cobra.Command{
+		Args:    cobra.ExactArgs(1),
+		Aliases: kindAliases[workloadKind],
+		Run: func(cmd *cobra.Command, args []string) {
+			enableOrDisableSource(cmd, args, workloadKind, disableInstrumentation)
+		},
+	}
+}
+
+func updateOrCreateSourceForObject(ctx context.Context, client *kube.Client, workloadKind k8sconsts.WorkloadKind, argName string, disableInstrumentation bool) (*v1alpha1.Source, error) {
+	var err error
+	obj := workload.ClientObjectFromWorkloadKind(workloadKind)
+	var objName, objNamespace, sourceNamespace string
+	switch workloadKind {
+	case k8sconsts.WorkloadKindDaemonSet:
+		obj, err = client.Clientset.AppsV1().DaemonSets(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindDeployment:
+		obj, err = client.Clientset.AppsV1().Deployments(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindStatefulSet:
+		obj, err = client.Clientset.AppsV1().StatefulSets(sourceNamespaceFlag).Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetNamespace()
+		sourceNamespace = sourceNamespaceFlag
+	case k8sconsts.WorkloadKindNamespace:
+		obj, err = client.Clientset.CoreV1().Namespaces().Get(ctx, argName, metav1.GetOptions{})
+		objName = obj.GetName()
+		objNamespace = obj.GetName()
+		sourceNamespace = obj.GetName()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var source *v1alpha1.Source
+	selector := labels.SelectorFromSet(labels.Set{
+		k8sconsts.WorkloadNameLabel:      obj.GetName(),
+		k8sconsts.WorkloadNamespaceLabel: sourceNamespace,
+		k8sconsts.WorkloadKindLabel:      string(workloadKind),
+	})
+	sources, err := client.OdigosClient.Sources(sourceNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+	if len(sources.Items) > 0 {
+		source = &sources.Items[0]
+	} else {
+		source = &v1alpha1.Source{
+			ObjectMeta: v1.ObjectMeta{
+				GenerateName: workload.CalculateWorkloadRuntimeObjectName(objName, workloadKind),
+				Namespace:    sourceNamespace,
+			},
+			Spec: v1alpha1.SourceSpec{
+				Workload: k8sconsts.PodWorkload{
+					Kind:      workloadKind,
+					Name:      objName,
+					Namespace: objNamespace,
+				},
+			},
+		}
+
+	}
+
+	source.Spec.DisableInstrumentation = disableInstrumentation
+
+	if len(sources.Items) > 0 {
+		source, err = client.OdigosClient.Sources(sourceNamespace).Update(ctx, source, v1.UpdateOptions{})
+	} else {
+		source, err = client.OdigosClient.Sources(sourceNamespace).Create(ctx, source, v1.CreateOptions{})
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if workloadKind == k8sconsts.WorkloadKindNamespace {
```

Comment:
nit: for a workload kinds, if the current instrumentation state is different than the one of the namespace, we can consider mentioning that here.

something like "NOTE: disabled instrumentation for workload Source, which exclude it from being instrumented by the namespace 'default' Source"